pipeline_dp/budget_accounting.py

```diff
@@ -100,7 +101,188 @@ def compute_budgets(self):
         for requested_budget in self._requested_budgets:
             eps = delta = 0
             if total_weight_eps:
-                eps = requested_budget.use_eps * self._eps * requested_budget.weight / total_weight_eps
+                numerator = requested_budget.use_eps * self._eps * requested_budget.weight
+                eps = numerator / total_weight_eps
             if total_weight_delta:
-                delta = requested_budget.use_delta * self._delta * requested_budget.weight / total_weight_delta
+                numerator = requested_budget.use_delta * self._delta * requested_budget.weight
+                delta = numerator / total_weight_delta
             requested_budget.budget.set_eps_delta(eps, delta)
+
+
+@dataclass
+class MechanismSpec:
+    """Specifies the parameters for a mechanism.
+
+    NoiseKind defines the kind of noise distribution.
+    noise is the minimized noise standard deviation.
+    """
+    noise_kind: NoiseKind
+    _noise_standard_deviation: float = None
+
+    @property
+    def noise(self):
+        """Noise value for the mechanism.
+
+        Raises:
+            AssertionError: The noise value is not calculated yet.
+        """
+        if self._noise_standard_deviation is None:
+            raise AssertionError(
+                "Noise standard deviation is not calculated yet.")
+        return self._noise_standard_deviation
+
+
+@dataclass
+class MechanismSpecInternal:
+    """Stores sensitivity and weight not exposed in MechanismSpec."""
```

Comment:
Nit: Please remove default arguments, anyway all those values should be provided in places where this class is used (they have default values in request_budget)